lib/pathresolver.js

```diff
@@ -86,7 +86,14 @@ PathResolver.prototype = {
     var parsedFrom = url.parse(mainDocUrl);
     var parsedTo = url.parse(absUrl);
     if (parsedFrom.protocol === parsedTo.protocol && parsedFrom.host === parsedTo.host) {
-      var pathname = pathPosix.relative(pathPosix.dirname(parsedFrom.pathname), parsedTo.pathname);
+      var parsedFromDir = pathPosix.dirname(parsedFrom.pathname);
+      // top-level file paths like `index.html` will return the dirname `.`.
+      // To calculate the relative path below, we should resolve against the
+      // root (`/`) instead.
+      if (parsedFromDir === '.') {
```

Comment:
For reference, the new relativeUrl function in v3 for reference.  https://github.com/Polymer/vulcanize/blob/v3/src/url-utils.ts#L48
https://github.com/Polymer/vulcanize/blob/v3/src/test/url-utils_test.ts#L67

None of tests address a case where pathname of the "from" URL isn't absolute, but I just tried calling the new relativeUrl with args 'index.html', 'some-other-thing.html' and it did the right thing, so I think we'll not see a regression in build when we switch to v3.